libraries/rush-lib/src/schemas/selector-expression.schema.json

```diff
@@ -0,0 +1,97 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "title": "JSON Selector Expression",
+  "description": "A JSON representation of selector expressions, which can be used to select sets of projects.",
+  "definitions": {
+    "expression": {
+      "oneOf": [
+        {
+          "$ref": "#/definitions/simpleSelector"
+        },
+        {
+          "$ref": "#/definitions/detailedSelector"
+        },
+        {
+          "$ref": "#/definitions/parameter"
+        },
+        {
+          "$ref": "#/definitions/unionOperator"
+        },
+        {
+          "$ref": "#/definitions/intersectOperator"
+        },
+        {
+          "$ref": "#/definitions/subtractOperator"
+        }
+      ]
+    },
+    "simpleSelector": {
+      "type": "string"
+    },
+    "detailedSelector": {
+      "type": "object",
+      "additionalProperties": false,
+      "requiredProperties": ["scope", "value"],
+      "properties": {
+        "scope": {
```

Comment:
In https://rushjs.io/pages/developer/selecting_subsets/, we say that in "--to X", the "X" is a _selector_, and that there are different _types_ of selectors.

We then describe a `tag:` selector, a `git:` selector, etc.

Perhaps words like "scope" and "provider" are just wishful thinking, and ways to avoid "selectorType" or simply `"selector": "git"`?

Reply:
@elliot-nelson this terminology arises from the grammar.  Your original prototype looked like this:
```
rush build --select 'to(tag:sdk and git:release/v3.0.0) and example-project'
```
...with the idea that Rush plugins might introduce new notations like `approver:team1`.

In this grammar, operators like `to()` and `and` use a simple standard tokenizer.  Whereas `release/v3.0.0` and `sdk` have arbitrary notations. When designing them, we need to be very careful that the syntax does not conflict with the grammar.  For example, `release/v3.0.0)` is a perfectly legal Git branch name, but is inexpressible in your notation, because the `)` will be misinterpreted as part of the `to()` function. (Yes yes, we can handle that by introducing escapes, or by banning unpleasant branch name choices, but awareness of this problem is my point.)

Beyond this parsing annoying, `release/v3.0.0` is also a learning curve for end users: Imagine a person doesn't know what Git is. They might wonder whether `/` is part of Rush's query language, or if it is part of the `git:` operator. In order to know, they need to learn the specialized syntax for Git branch names.

So, if you look at the docs, the word "selector" was really introduced to mean "AST leaf nodes with their own weird syntaxes you have to learn." The island grammars waiting to break our parser if we aren't careful.

`--to` selects a set of projects, but it is not a "selector" because `--to` has standard well-behaved syntax.  It is a grammatically normal friendly selection "operator" or "function", being represented as as a CLI "parameter".